### PR TITLE
docs: Readme - Deprecate 3.x

### PR DESCRIPTION
**Pull request by fabio-looker**
Adds notes to readme clarifying that 3.x is deprecated, 4.0 is stable, and linking to 3.0 deprecation notice.

### CODE REVIEW

**Review by josephaxisa — APPROVED** [review]
lgtm. I'd wait for John to review as well before merging

File: README.md
Comment (jkaster) [inline README.md:32]: The API Docs viewer is gone now, and has been replaced by API Explorer, which is available to install for each instance and also on the developer portal

File: README.md
Comment (jkaster) [inline README.md:40]: I'm pretty sure we've dropped 3.1 from Kotlin and Go as well. We should also drop it from Look# if we haven't yet because no one should use API 3.x on a statically typed language.

File: README.md
Comment (jkaster) [inline README.md:57]: If 3.1 is gone from Kotlin, you can make it "Not applicable" as well.

File: README.md
Comment (jkaster) [inline README.md:180]: Did you actually test this? I haven't tried the legacy generator in literally a year or more.

File: README.md
Comment (jkaster) [inline README.md:285]: I wonder where we're actually using this now. Worth spending a few minutes verifying whether we still use it or not.

**Review by jkaster — CHANGES_REQUESTED** [review]
Great job sweeping through the docs. This will be a good update.

File: README.md
Comment (fabio-looker) [reply README.md:32]: Updated to API Explorer

File: README.md
Comment (fabio-looker) [reply README.md:40]: I've cleaned this up a bit by focusing the language primarily around the "Looker supported" SDKs

File: README.md
Comment (fabio-looker) [reply README.md:57]: Updated all strongly typed SDKs to 3.1 "Not supported"

File: README.md
Comment (fabio-looker) [reply README.md:180]: I tested it and it threw an opaque error from sdk-rtl/lib/transport.js:221... although I couldn't immediately tell if this was a resolvable network/instance/config issue, it seems like your comment is suggesting that this section should be removed, so I've tentatively removed it.

File: README.md
Comment (fabio-looker) [reply README.md:285]: Actually used in only 1 place:
- In a "check'' against the API login endpoint during a "wait for looker" github script.

Programmatically passed-through / typed in 2 places
- Docker config passes it around
- Go type delarations include it

Doc'd in 4 places:
- This main readme
- Two env sample files
- A Python notebook example

From the one use, I think the default value (4.0) should be fine (and indistinguishable in functionality from 3.1) in virtually every situation besides trying to build against a _very_ old Looker version, so maybe we leave the variable there but remove it from the documentation?

File: README.md
Comment (jkaster) [reply README.md]: I'm good with removing it

File: README.md
Comment (jkaster) [reply README.md:273]: I think removing it from the doc is a good step

**Review by jkaster — COMMENTED** [review]
Good updates. I think there's one pending removal you called out, but this is looking good so far.

File: README.md
Comment (jkaster) [inline README.md:40]:
```suggestion
For SDKs that support multiple API versions, there will be `methods.*` and `models.*` collections generated for each API version. Each API version is exposed under a distinct class name from which to instantiate an initial SDK object.
```

File: README.md
Comment (jkaster) [inline README.md:173]: We should keep this section in because the legacy generator can be used to generate language SDKs we don't have.

**Review by jkaster — CHANGES_REQUESTED** [review]
We're very close

File: README.md
Comment (fabio-looker) [reply README.md]: I added the section back, but omitted the commentary around which SDKs are "currently" available and around changing the API versions (the looker-sample.ini file still explains the apiVersions property with a comment in that file)

**Review by jkaster — APPROVED** [review]
Looks great!

**Pull request merged**
